### PR TITLE
feat: add project API

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/cyclone/v1alpha1/project.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Project defines a project which holds a lot of common default information of some workflow under it.
+type Project struct {
+	// Metadata for the resource, like kind and apiversion
+	meta_v1.TypeMeta `json:",inline"`
```
Comment (supereagle) [inline pkg/apis/cyclone/v1alpha1/project.go]: Will we support alias, description of project?

File: pkg/apis/cyclone/v1alpha1/project.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	// Metadata for the particular object, including name, namespace, labels, etc
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the Workflow specification
+	Spec ProjectSpec `json:"spec"`
+}
+
+// ProjectSpec defines project specification.
+type ProjectSpec struct {
+	// Services contains default value of various type of integrations.
+	Services []ServiceItem `json:"services"`
```
Comment (supereagle) [inline pkg/apis/cyclone/v1alpha1/project.go]: `service` is not very good as it is easy confused with k8s service and dependent services in integration test.

Comment (cd1989) [reply pkg/apis/cyclone/v1alpha1/project.go]: Indeed, any suggestions ?

File: pkg/apis/cyclone/v1alpha1/project.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	core_v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Project defines a project which holds a lot of common default information of some workflow under it.
```
Comment (cd1989) [inline pkg/apis/cyclone/v1alpha1/project.go]: // Project defines a project which holds common information of workflows under it.

**Pull request merged**